File: 250103/그래프 탐색/graph-traversal.cpp
#include <iostream>
#include <vector>
#define MAX_N 1000
using namespace std;

int n, m;
int vertex_cnt;

vector<int> graph[MAX_N+1];
bool visited[MAX_N+1];

void DFS(int vertex) {
    for(int i=0; i<graph[vertex].size(); i++) {
        int curr_v = graph[vertex][i];
        if(!visited[curr_v]) {
            visited[curr_v] = true;
            vertex_cnt++;
            DFS(curr_v);
        }
    }
}

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n>>m;

    int v1, v2;
    for(int i=0; i<m; i++){
        cin >> v1>>v2;
        graph[v1].push_back(v2);
        graph[v2].push_back(v1);
    }

    visited[1] = true;
    DFS(1);

    cout << vertex_cnt;
    return 0;
}

File: 250128/그래프 탐색/graph-traversal.py
n, m = tuple(map(int, input().split()))

graph = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)]

vertex_cnt = 0

def dfs(vertex):
    global vertex_cnt

    for cur_v in graph[vertex]:
        if not visited[cur_v]:
            visited[cur_v] = True
            vertex_cnt += 1
            dfs(cur_v)

for _ in range(m):
    v1, v2 = tuple(map(int, input().split()))

    graph[v1].append(v2)
    graph[v2].append(v1)

visited[1] = True
dfs(1)

print(vertex_cnt)